vic/drivers/python/setup.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import os
+import re
+import sys
+import sysconfig
+import warnings
+import glob
+import subprocess
+import shutil
+
+from setuptools import setup, find_packages, Command
+from setuptools.extension import Extension
+
+MAJOR = 5
+MINOR = 0
+MICRO = 0
+ISRELEASED = False
+VERSION = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
+QUALIFIER = ''
+
+FULLVERSION = VERSION
+write_version = False
+
+vic_root_rel_path = os.path.join(os.pardir, os.pardir)
+vic_root_abs_path = os.path.abspath(vic_root_rel_path)
+
+
+start_dir = os.path.abspath(os.getcwd())
+setup_dir = os.path.dirname(os.path.abspath(__file__))
+os.chdir(setup_dir)
+
+
+class CleanCommand(Command):
+    """Custom clean command to tidy up the project root."""
+    user_options = []
+
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        for path in ['./build', './dist', './__pycache__', './vic.egg-info']:
+            try:
+                print('removing %s' % path)
+                shutil.rmtree(path)
+            except:
+                pass
+
+        files = ['vic/_vic.py']
+        files.extend(glob.glob('vic/*pyc'))
+        files.extend(glob.glob('vic_core*'))
+
+        for filename in files:
+            try:
+                print('removing %s' % filename)
+                os.remove(filename)
+            except:
+                pass
+
+
+# -------------------------------------------------------------------- #
+def write_version_py(filename=None):
+    cnt = """\
+version = '%s'
+short_version = '%s'
+"""
+    if not filename:
+        filename = os.path.join(setup_dir, 'vic', 'version.py')
+
+    a = open(filename, 'w')
```

Comment:
shouldn't you try on `open` rather than or `write`? 

Or use:
```python
    with open(filename, 'w') as f:
         f.write(cnt % (FULLVERSION, VERSION))
```

From python documentation (https://docs.python.org/2/tutorial/inputoutput.html): 
_It is good practice to use the with keyword when dealing with file objects. This has the advantage that the file is properly closed after its suite finishes, even if an exception is raised on the way. It is also much shorter than writing equivalent try-finally blocks._

Lastly: I assume you are not using newer strong formatting using `{}` to remain compatible with older python versions? If not, then I would standardize on py3 string formatting.